src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/MemoryLayout.java

```diff
@@ -291,6 +285,41 @@ default long byteAlignment() {
      */
     MemoryLayout withBitAlignment(long bitAlignment);
 
+    /**
+     * Returns the attribute with the given name if it exists, or an empty optional
+     *
+     * @param name the name of the attribute
+     * @return the optional attribute
+     */
+    Optional<Constable> attribute(String name);
+
+    /**
+     * Returns the attribute with the given name and the given type if it exists, or an empty optional
+     *
+     * @param name the name of the attribute
+     * @param type the type to filter by
+     * @return the optional attribute
+     */
+    default <T extends Constable> Optional<T> attribute(String name, Class<T> type) {
```

Comment:
Not sure I like this method. While it works assuming all constable implementation are non-generic, it will stop working correctly as soon as you have a generic class implementing a constable. Maybe save the sugar for later?

Reply:
I see what you mean. It's not that easy to get a `Class<T>` when `T` is a generic type. I'll remove this helper for now. (adding SystemABI.Type::fromLayout removed most use-cases we currently have as well).